123-job-jwt/proposal.md

```diff
@@ -0,0 +1,137 @@
+# Summary
+
+Mint a signed JSON Web Token (JWT) and deliver to the job runtime container.
+Additionally, provide a validation endpoint to allow for external or internal
+validation of the issued JWT.
+
+## Motivation
+
+Concourse runners lack an identity mechanism that can be used to authenticate to
+external systems. This lack means that an initial secret must be provided to
+each pipeline in Concourse in a manual manner. Many other CI solutions and cloud
+providers solve the "secret zero" problem by provisioning a signed JWT to their
+workloads automatically, and providing a path to signature verification in a
+manner that the external applications can trust.
+
+Should this be implemented, Concourse jobs will be able to leverage this
+automatically generated JWT to be the "secret zero" needed to communicate with
+external systems (such as Hashicorp Vault, AWS, and others) and will no longer
+require a Concourse administrator (or person configuring a pipeline) to provide
+a static credential to Concourse. This alleviates the need for a human to handle
+sensitive credentials, increases security by ensuring credentials are not
+statically configured and JWTs are ephemeral to each run, and reduces
+Concourse's need to properly handle and encrypt data (can be limited to storing
+the JWT private signing key).
+
+## Proposal
+
+Adopt [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519) and a Go library
+that [implements](https://github.com/golang-jwt/jwt) it. Additionally adopt
+[RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517) and a Go library that
+implements it. (Perhaps [jwx](https://github.com/lestrrat-go/jwx)?)
```

Comment:
We should add RFC8414 here.

https://www.rfc-editor.org/rfc/rfc8414.html#section-3 specifies the `/.well-known/openid-configuration` endpoint, and also points to the JWKS URL.

Reply:
Absolutely. Supporting OIDC-Discovery is a must-have here